Docs: overview of the application logic

thekaveman opened the pull request:
Based on some recent discussion and onboarding new team members, this page provides a high-level overview of all of the main phases through which the user is taken by the app.

Preview page: https://benefits-2139--cal-itp-previews.netlify.app/development/application-logic/

lalver1 commented on docs/development/application-logic.md:
```diff
@@ -0,0 +1,287 @@
+## Identity proofing
+
+In this phase, Cal-ITP Benefits takes the user through an [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/)
+flow as a Client (RP) of the CDT Identity Gateway (the Identity Provider or IDP), via Login.gov.
```
```suggestion
flow as a Client (the Relying Party or RP) of the CDT Identity Gateway (the Identity Provider or IDP), via Login.gov.
```

lalver1 reviewed (commented):
This looks great! 👍  I think the `Entrypoint` and `Key supporting files` call-outs work well to link the descriptions in the documentation with the code. The last diagram in the [Enrollment](https://benefits-2139--cal-itp-previews.netlify.app/development/application-logic/#enrollment) section is also very useful given the different tokens used in this phase.
I only added one minor suggested change.

thekaveman commented:
Thanks for the suggestion @lalver1, I added that in.

I also added to each of the diagrams a block for the interaction with analytics, it's helpful to see when the key events are generated in each phase.

lalver1 commented:
I agree, adding the interaction with analytics made the diagrams even more useful yet they still look clear.

lalver1 reviewed (approved):
Looks good 👍

angela-tran reviewed (approved):
This looks great! I really like the diagrams.
